web/src/containers/StateDashboardCurrentDocuments.js

```diff
@@ -0,0 +1,205 @@
+import PropTypes from 'prop-types';
+import React, { Component, Fragment } from 'react';
+import { connect } from 'react-redux';
+
+import { selectApd } from '../actions/apd';
+import Btn from '../components/Btn';
+import Icon, {
+  faArrowRight,
+  faCheckCircle,
+  faCircle,
+  faEdit,
+  faSpinner
+} from '../components/Icons';
+
+const ProgressDot = ({ done, started, text }) => {
+  let color = '#aaaaaa';
+  if (done) {
+    color = '#9dd887';
+  } else if (started) {
+    color = '#124f81';
+  }
+
+  let labelColor = '';
+  if (!done && !started) {
+    labelColor = ' light gray';
+  }
+
+  return (
+    <div className="center">
+      <Icon icon={done ? faCheckCircle : faCircle} color={color} size="2x" />
+      <div className={`absolute pl1 ${labelColor}`}>{text}</div>
+    </div>
+  );
+};
+
+ProgressDot.propTypes = {
+  done: PropTypes.bool,
+  started: PropTypes.bool,
+  text: PropTypes.bool
+};
+
+ProgressDot.defaultProps = {
+  done: false,
+  started: false,
+  text: null
+};
+
+const ProgressLine = ({ done }) => (
+  <div
+    className={`flex-auto bg-${done ? 'blue' : 'gray'}`}
+    style={{ height: '3px' }}
+  />
+);
+
+ProgressLine.propTypes = {
+  done: PropTypes.bool
+};
+
+ProgressLine.defaultProps = {
+  done: false
+};
+
+const DocumentItem = ({ name, status, buttonClick }) => {
+  const states = {
+    one: {
+      done: false,
+      started: true,
+      title: 'Submitted'
+    },
+    two: {
+      done: false,
+      started: false,
+      title: 'Reviewed'
+    },
+    three: {
+      done: false,
+      started: false,
+      title: 'Approved'
+    }
+  };
+
+  let actionButtonText = 'View';
+  let actionButtonIcon = faArrowRight;
+
+  switch (status) {
+    case 'draft':
+      states.one.title = 'Drafting';
+      actionButtonText = 'Open';
+      actionButtonIcon = faEdit;
+      break;
+
+    case 'in review':
+      states.one.done = true;
+      states.two.started = true;
+      states.two.title = 'Reviewing';
+      break;
+
+    case 'reviewed':
+      states.one.done = true;
+      states.two.done = true;
+      break;
+
+    case 'approved':
+      states.one.done = true;
+      states.two.done = true;
+      states.three.done = true;
+      actionButtonText = 'Approval letter';
+      break;
+
+    default:
+      break;
+  }
```

Comment:
I'm really not wild about this `switch` statement to map a status into state info, but I wasn't sure offhand of a better way...

Reply:
The only thing that comes to mind is creating an object that maps these to the status data, like:
```
words = {
  'draft': {
    'actionButtonText': 'View',
    'actionButtonIcon': faEdit,
  }
}
```
and then you could just return something like `{ words[status].actionButtonText, words[status].actionButtonIcon }`. I'm not sure it's _better_, just different.

The other thing that might come in really handy is a `constants` file where you can define all your text. CALC does this [here](https://github.com/18F/calc/blob/develop/frontend/source/js/data-explorer/constants.js), and it's saved me a world of typo pain. (Also makes it _super_ easy when someone decides that text needs to be tweaked slightly.)

Reply:
Oooh, I like it.  This button text will eventually come from a resource file, but the key can live in a constants file.  I'll make that change!

For the statuses, those are read-only from the API, so we could massage them into `Symbol`s also defined in a constants file, to make more guarantees about using them correctly elsewhere.  🤔  Probably shouldn't tack more onto this PR, but I love this as a future cleanup.  The statuses aren't used much right now, so good time to make that change.